node/services/service.go

```diff
@@ -26,8 +29,13 @@ func Header(lc fx.Lifecycle, ps *pubsub.PubSub) *header.Service {
 }
 
 // Block constructs new block.Service.
-func Block(lc fx.Lifecycle, fetcher block.Fetcher, store ipld.DAGService) *block.Service {
-	service := block.NewBlockService(fetcher, store)
+func Block(
+	lc fx.Lifecycle,
+	fetcher block.Fetcher,
+	store ipld.DAGService,
+	broadcaster header.Broadcaster,
+) *block.Service {
+	service := block.NewBlockService(fetcher, store, broadcaster)
```

Comment:
It is ok for now, but I would feed it with the whole header service instead of Broadcaster. The reason is that we won't have multiple Broadcaster and broadcasting will happen only in one specific place, so it seems to me like there is no point in abstraction. I know that you want to make it similar to Subscriber, like if there is a Subscriber then there should be Publisher respectively.

Reply:
Note, I am ok to leave it like this for now.

Reply:
Yeah I don't think that block service needs access to the whole header service -- it gets only the instance of Broadcaster from the header service btw - it doesn't instantiate its own. Do you think it should and then only get fed pubsub from the DI app?

Reply:
I just don't think we need Broadcaster. Passing HeaderService to BlockService is temporary. This will be fixed once we start listen for Headers first from Core and not for Blocks